### ISSUE TITLE
Open Bookmarks in New Tab?

### ISSUE THREAD

**Issue by DaveH1** [Alex313031/Thorium-Win#209]
As a user of 360Chome, one of the most useful things in that browser was the setting to always open bookmarks in a new tab.
Could this be added as an option to Thorium?
Thanks, Dave.

**Comment by Unknown78** [comment]
Just click the link with middle scroll button if you have a mouse. lol.

**Comment by DaveH1** [comment]
Yes, I'm well aware of that option thanks, and the right click menu option too!
It's just something I've got used to happening automatically with browsers I've previously used.
It's there as a settings option in 360Chrome, so presumably wouldn't be that difficult to provide in Thorium.
My other two preferred options, not closing the browser when the last tab is closed, and double-clicking to close a tab, are already there. It's just opening bookmarks in a new tab, which is still missing!

**Comment by Alex313031** [comment]
@DaveH1 I know how to make Chromium register different mouse button presses. I use this, for example, in the flag to enable closing a tab by right clicking on it. The issue is finding where in the source code it decides to open a bookmark in a new tab on middle click press.

**Comment by DaveH1** [comment]
Thanks @Alex313031.
Hopefully this can be added at some point, but it's obviously not a high priority!
Cheers, Dave.

**Comment by JoachimHenze** [comment]
@DaveH1 you could either use the chrome-flag #custom-ntp and set that to chrome://bookmarks
Or @Alex313031 could pick the implementation that the Supermium developers added here for me https://github.com/win32ss/supermium/issues/238
That retores the behavior of Chrome 49 to display the bookmark bar only for new tabs but hides it for tabs which do already display webpage content (to save vertical screen space).

Not sure which of boith you do actually request for here.

**Comment by DaveH1** [comment]
Thanks @JoachimHenze but that's not quite what I'm asking about.
What I'm asking for is an option (it is in the settings of 360Chrome) to always open them in a new tab when bookmarks are selected, instead of replacing the existing foreground tab.

![Clipboard-3](https://github.com/Alex313031/Thorium-Win/assets/6755753/cad537ca-95fd-473e-8fa4-74ae6e1d2b14)

This is especially good (for me anyway) when selecting entries on the bookmarks bar.

**Comment by gz83** [comment]
Here is a GitHub project that may be suitable for you to use. In fact, I am willing to try to implement the functions you want from the source code level, but my current main focus is to assist Google in modifying and improving the webgpu and Android codes.

https://github.com/Bush2021/chrome_plus

@DaveH1

**Comment by DaveH1** [comment]
Thanks @gz83 that looks like a very interesting project!
However, the function I'm asking about must be already capable of being implemented in Chromium as it's already there in 360Chrome, which is based on Chromium 86.

**Comment by gz83** [comment]
What I proposed to you earlier is a transitional solution.

@DaveH1

**Comment by DaveH1** [comment]
@gz83 
Yes of course.
Alex was going to look into adding this as an option in Thorium, I think it was a matter of isolating the code used when the middle click is used to open a link in a new tab.
I have downloaded the option you kindly linked to, and I will give it a try.
Thanks very much.

**Comment by DaveH1** [comment]
@gz83 
Hi again! Off-topic, but have you actually used the chrome_plus option?
I can't see how to use it.
It says you just have to put the version.dll into the same folder as chrome.dll, but there's obviously more to it than that!
It comes with three folders, an 'app' folder containing version.dll and chrome++.ini, the settings file, and two other folders, 'Cache' and 'Data' both of which are empty.
What are you supposed to do with them, and the settings ini file?
I've tried various options, but nothing I've tried has actually made it work!
Thanks, Dave.

**Comment by gz83** [comment]
I have used it before and it worked fine. From your description, it seems that you need to look carefully at the introduction in that repo.
@DaveH1

**Comment by DaveH1** [comment]
@gz83 
Thanks, but I really can't see what I'm missing here.

![Clipboard-1](https://github.com/Alex313031/Thorium-Win/assets/6755753/bf8a32d1-7565-47bd-a0ac-b46ff4c092a7)

In the introduction, under Installation, it just says "Place version.dll in the same directory as chrome.exe" and that's all it says!
Where are the other instructions?
Cheers, Dave.

**Comment by gz83** [comment]
Regarding how to use this software, it is recommended to consult the author and other users in the repo of this software.

@DaveH1

**Comment by DaveH1** [comment]
Thanks, but can't you just tell me?!
I only want to test it.
Yes, this is completely off-topic here.

**Comment by gz83** [comment]
Please place the chrome++ file like this.

![image](https://github.com/Alex313031/Thorium-Win/assets/17492865/2540f7c5-7328-41ba-beb7-51091e5c68a8)

Next, you need to modify the chrome++.ini file, which uses 0 or 1 to control the turning off or on of the function.

Every time you modify the chrome++.ini file, you need to restart the browser for the changes to take effect.

Finally, your browser user data may be reset, please make sure to back it up.

@DaveH1

**Comment by DaveH1** [comment]
Thanks very much!
I have put the files into the folder Program Files\Thorium\Chrome\Application, which is where yours seem to be.
This is different to the instructions, which say that version.dll should be in the same folder as Chrome.dll.
Still nothing happens, no matter what I put into chrome++.ini.
Sorry, I should have stated that I'm using Thorium Legacy on Windows XP.
Does that make a difference? Perhaps this just won't work on that version under that OS.

**Comment by gz83** [comment]
Oops, I forgot you were using Windows XP, and Chrome++ is not currently available on Windows XP.

In addition, I will move this issue to the legacy repo to reduce misunderstandings.

@DaveH1
